File: client/src/app/core/models/requests/update-project-task-request.ts
import { EstimateType } from '@core/enums/estimate-type';
import { TaskPriority } from '@core/enums/task-priority';

export interface UpdateProjectTaskRequest {
  id?: number;
  name: string;
  description: string;
  statusId?: number;
  sortOrder?: number;
  ownerId?: string | null;
  assigneeIds?: string[];
  tags?: string[];
  projectId?: number;
  priority?: TaskPriority | null;
  estimateType?: EstimateType | null;
  estimateValue?: number | null;
  startDate?: string | null;
  dueDate?: string | null;
}

File: server/Netptune.Core/Requests/UpdateProjectTaskRequest.cs
using System.Text.Json.Serialization;
using Netptune.Core.Enums;

namespace Netptune.Core.Requests;

public record UpdateProjectTaskRequest
{
    private DateOnly? startDate;

    private DateOnly? dueDate;

    public int Id { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public int? StatusId { get; set; }

    public double? SortOrder { get; set; }

    public string? OwnerId { get; set; }

    public List<string>? AssigneeIds { get; set; }

    public List<string>? Tags { get; set; }

    public TaskPriority? Priority { get; set; }

    public EstimateType? EstimateType { get; set; }

    public decimal? EstimateValue { get; set; }

    public DateOnly? StartDate
    {
        get => startDate;
        set
        {
            startDate = value;
            StartDateSpecified = true;
        }
    }

    [JsonIgnore]
    public bool StartDateSpecified { get; private set; }

    public DateOnly? DueDate
    {
        get => dueDate;
        set
        {
            dueDate = value;
            DueDateSpecified = true;
        }
    }

    [JsonIgnore]
    public bool DueDateSpecified { get; private set; }
}
